mac-setup/build-app.sh

```diff
@@ -1,10 +1,18 @@
 #!/bin/bash
-export PATH="$HOME/.new_local/bin:$PATH"
+
+set -e
 
 ## Mac Setup script
 ## 1. do the build, will not be called from this script
 ## 2. call this script
 ## 3. an .app will be packed
+#
+# This script should be run from a jhbuild environment
+
+if [ -z "$UNDER_JHBUILD" ]; then
+  echo 'Error: this command needs to be run within jhbuild.'
+  exit 1
+fi
```

Comment:
We still use `build-app.sh` on Azure Pipelines, where we have no `jhbuild`.

Reply:
That's a good point.

It might be good to also add instructions on how to create a binary blob. This would be convenient for building the dependencies on a special jhbuild user account, and then using the regular user account to build the final xournalpp executable & bundle.

Reply:
Fortunately that's quite simple nowadays: Just download the script https://raw.githubusercontent.com/xournalpp/xournalpp-pipeline-dependencies/base/gtk/package-gtk-bin.sh and run it via
```term
./package-gtk-bin.sh "$HOME"/gtk
```
Extract the split up archive via
```term
cat gtk-bin.tar.gz.* | tar -xz
```

Reply:
After creating a binary blob with a special jhbuild user account and trying to use it on a regular user account, I think this is too difficult. There's too many paths that can get messed up by moving the gtk prefix, so we'll have to assume that the app is built with the special jhbuild user account.

I think we can at least document how the blob build works though.

Reply:
I changed this check so that it only prints out a non-fatal warning instead of quitting.